packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -0,0 +1,272 @@
+  /** x-axis range:  */
+  independentAxisRange?: NumberRange;
+  /** y-axis range: */
+  dependentAxisRange?: NumberRange;
+  /**
+   * Array of size 2 that contains a label for the left and right side
+   * of the x axis. (Not yet implemented). Expect this to be passed by the viz based
+   * on the type of data we're using (genes vs taxa vs etc.)
+   */
+  comparisonLabels?: Array<string>;
+  /** What is this plot's name? */
+  plotTitle?: string;
+  /** marker color opacity: range from 0 to 1 */
+  markerBodyOpacity?: number;
+}
+
+const EmptyVolcanoPlotData: VolcanoPlotData = {
+  foldChange: [],
+  pValue: [],
+  adjustedPValue: [],
+  pointId: [],
+};
+
+interface DataPoint {
+  foldChange: string;
+  pValue: string;
+  adjustedPValue: string;
+  pointId: string;
+  color: string;
+}
+
+/**
+ * The Volcano Plot displays points on a (magnitude change) by (significance) xy axis.
+ * It also colors the points based on their significance and magnitude change.
+ */
+function VolcanoPlot(props: VolcanoPlotProps) {
+  const {
+    data = EmptyVolcanoPlotData,
+    independentAxisRange,
+    dependentAxisRange,
+    significanceThreshold,
+    log2FoldChangeThreshold,
+    markerBodyOpacity,
+    ...restProps
+  } = props;
+
+  /**
+   * Find mins and maxes of the data and for the plot
+   */
+
+  const dataXMin = min(data.foldChange.map(Number));
+  const dataXMax = max(data.foldChange.map(Number));
+  const dataYMin = min(data.pValue.map(Number));
+  const dataYMax = max(data.pValue.map(Number));
+
+  // Determine mins, maxes of axes in the plot.
+  // These are different than the data mins/maxes because
+  // of the log transform and the little bit of padding.
+  //
+
+  let xMin: number;
+  let xMax: number;
+  let yMin: number;
+  let yMax: number;
+
+  // Log transform for plotting, and add a little margin for axes
+  if (dataXMin && dataXMax) {
+    xMin = Math.log2(dataXMin);
+    xMax = Math.log2(dataXMax);
+    xMin = xMin - (xMax - xMin) * 0.05;
+    xMax = xMax + (xMax - xMin) * 0.05;
+  } else {
+    xMin = 0;
+    xMax = 0;
+  }
+  if (dataYMin && dataYMax) {
+    yMin = -Math.log10(dataYMax);
+    yMax = -Math.log10(dataYMin);
+    yMin = yMin - (yMax - yMin) * 0.05;
+    yMax = yMax + (yMax - yMin) * 0.05;
+  } else {
+    yMin = 0;
+    yMax = 0;
+  }
+
+  /**
+   * Turn the data (array of arrays) into data points (array of points)
+   */
+
+  let dataPoints: DataPoint[] = [];
+
+  // Loop through the data and return points. Doesn't really matter
+  // which var of the data we map over.
+  data.foldChange.forEach((fc, ind: number) => {
+    dataPoints.push({
+      foldChange: fc,
+      pValue: data.pValue[ind],
+      adjustedPValue: data.adjustedPValue[ind],
+      pointId: data.pointId[ind],
+      color: assignSignificanceColor(
+        Math.log2(Number(fc)),
+        Number(data.pValue[ind]),
+        significanceThreshold,
+        log2FoldChangeThreshold,
+        significanceColors
+      ),
+    });
+  });
```

Comment:
**fun fact**: this totally works! For some food for thought, some folks would write it like this:

```js
  const dataPoints: DataPoint[] = data.foldChange.reduce((datapoints: DataPoint[], fc, ind) => {
    return [
      ...datapoints,
      {
        foldChange: fc,
        pValue: data.pValue[ind],
        adjustedPValue: data.adjustedPValue[ind],
        pointId: data.pointId[ind],
        color: assignSignificanceColor(
          Math.log2(Number(fc)),
          Number(data.pValue[ind]),
          significanceThreshold,
          log2FoldChangeThreshold,
          significanceColors
        ),
      },
    ];
  }, []);
```